### ISSUE TITLE
FIXME: Runner Progress Bar

### ISSUE THREAD

**Issue by hk21702** [RimSort/RimSort#393]
**Describe the bug:**
Runner progress bar state currently does not change, and the “success” state is undefined as well. 

Best-case fix, it should be changed such that it is dynamic with the steps reported, but it should at least actually correctly report the success status.

- OS: All
- RimSort version: v1.0.8 (Present in older versions as well)
